src/west/cmd/project.py

```diff
@@ -528,35 +527,35 @@ def _rebase(project):
     _git(project, 'rebase (manifest-rev-branch)')
 
 
-def _cloned(project):
-    # Returns True if the project's repository exists and looks like a Git
-    # repository.
+def _cloned(project, verify_repo=False):
+    # Returns True if the project's path is a directory that looks
+    # like the top-level directory of a Git repository, and False
+    # otherwise.
     #
-    # Prints a warning if the project's clone path exist but doesn't look like
-    # a Git repository.
+    # verify_repo (default: False):
+    #   If True, an error is raised instead of returning False.
+
+    def handle(result):
+        log.dbg('project', project.name,
+                'is {}cloned'.format('' if result else 'not '),
+                level=log.VERBOSE_EXTREME)
+        if not result and verify_repo:
+            _die(project,
+                 '(name-and-path) is not the top-level directory of a '
+                 'Git repository')
+        return result
+
+    if not os.path.isdir(project.abspath):
+        return handle(False)
 
-    if os.path.exists(project.abspath):
-        _verify_repo(project)
-        return True
-
-    return False
-
-
-def _verify_repo(project):
-    # Raises an error if the project's clone path is not the top-level
-    # directory of a Git repository
-    log.dbg('verifying project:', project.name, level=log.VERBOSE_EXTREME)
     # --is-inside-work-tree doesn't require that the directory is the top-level
     # directory of a Git repository. Use --show-cdup instead, which prints an
     # empty string (i.e., just a newline, which we strip) for the top-level
     # directory.
     res = _git(project, 'rev-parse --show-cdup', capture_stdout=True,
                check=False)
 
-    if res.returncode or res.stdout:
-        _die(project, '(name-and-path) is not the top-level directory of a '
-                      'Git repository, as reported by '
-                      "'git rev-parse --show-cdup'")
+    return handle(not(res.returncode or res.stdout))
```

Comment:
`not` is an operator and not a function, so could make it `not (res.returncode or res.stdout)`.

Reply:
Thank you. I'm surprised flake8 did not complain.